dev_tools/notebooks/isolated_notebook_test.py

```diff
@@ -195,15 +201,47 @@ def test_notebooks_against_released_cirq(partition, notebook_path, base_env):
         raise_on_fail=False,
         out=shell_tools.TeeCapture(),
         err=shell_tools.TeeCapture(),
+        # important to get rid of PYTHONPATH specifically, which contains
+        # the Cirq repo path due to check/pytest
+        env={},
     )
 
     if status != 0:
         print(stderr)
         pytest.fail(
             f"Notebook failure: {notebook_file}, please see {out_path} for the output "
             f"notebook (in Github Actions, you can download it from the workflow artifact"
-            f" 'notebook-outputs')"
+            f" 'notebook-outputs'). \n"
+            f"If this is a new failure in this notebook due to a new change, "
+            f"that is only available in master for now, consider adding `pip install --pre cirq` "
+            f"instead of `pip install cirq` to this notebook, and exclude it from "
+            f"dev_tools/notebooks/isolated_notebook_test.py."
         )
 
     if rewritten_notebook_descriptor:
         os.close(rewritten_notebook_descriptor)
+
+
+def _unreleased_notebooks():
+    return functools.reduce(
+        lambda a, b: a.union(b),
+        list(set(glob.glob(g, recursive=True)) for g in NOTEBOOKS_DEPENDING_ON_UNRELEASED_FEATURES),
```

Comment:
Good point, I thought first that it would be useful, but I guess we can introduce it later as needed. Typically we exclude notebooks on a one by one basis so this should be sufficient.